pkg/tests/cross-tests/cross_test.go

```diff
@@ -403,3 +404,122 @@ func TestAws2442(t *testing.T) {
 		Config2:  cfg2,
 	})
 }
+
+func TestSimpleOptionalComputedNoChange(t *testing.T) {
+	skipUnlessLinux(t)
+	config := tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{})
+	runDiffCheck(t, diffTestCase{
+		Resource: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:     schema.TypeString,
+					Optional: true,
+					Computed: true,
+				},
+			},
+			CreateContext: func(
+				ctx context.Context, rd *schema.ResourceData, i interface{},
+			) diag.Diagnostics {
+				err := rd.Set("name", "ComputedVal")
+				require.NoError(t, err)
+				rd.SetId("someid")
+				return make(diag.Diagnostics, 0)
+			},
+		},
+		Config1: config,
+		Config2: config,
+	})
+}
+
+func TestOptionalComputedAttrCollectionNoChange(t *testing.T) {
+	skipUnlessLinux(t)
+	config := tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{})
+
+	for _, tc := range []struct {
+		name     string
+		maxItems int
+		typ      schema.ValueType
+	}{
+		{"list", 0, schema.TypeList},
+		{"set", 0, schema.TypeSet},
+		{"list max items one", 1, schema.TypeList},
+		{"set max items one", 1, schema.TypeSet},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			runDiffCheck(t, diffTestCase{
+				Resource: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"security_groups": {
+							Type:     tc.typ,
+							Optional: true,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							MaxItems: tc.maxItems,
+						},
+					},
+					CreateContext: func(
+						ctx context.Context, rd *schema.ResourceData, i interface{},
+					) diag.Diagnostics {
+						err := rd.Set("security_groups", []string{"sg1"})
+						require.NoError(t, err)
+						rd.SetId("someid")
+						return make(diag.Diagnostics, 0)
+					},
+				},
+				Config1: config,
+				Config2: config,
+			})
+		})
+	}
+}
+
+func TestOptionalComputedBlockCollectionNoChange(t *testing.T) {
+	skipUnlessLinux(t)
+	config := tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{})
```

Comment:
I keep being thrown off by these not being valid (not specifying the Type), but perhaps since it drives the serializer only that's OK as specifying the type would not help.

Reply:
Yeah, actually specifying the type correctly doesn't work, since it then errors out if you don't specify an attribute of the object - the type in NewValue needs to match the value, not the schema I believe.